mmv1/products/apigee/EnvironmentAddonsConfig.yaml

```
@@ -0,0 +1,66 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'EnvironmentAddonsConfig'
+api_resource_type_kind: AddonsConfig
+description: Enable/Disable add-ons for an Apigee environment.
+references:
+  guides:
+    'Enable Analytics Add-On': 'https://cloud.google.com/apigee/docs/api-platform/reference/manage-analytics-add-on'
+    'Enable ApiSecurity Add-On': 'https://cloud.google.com/apigee/docs/api-platform/reference/manage-security-add-on'
+  api: 'https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.environments.addonsConfig/setAddonEnablement'
+docs:
+base_url: '{{env_id}}/addonsConfig'
+self_link: '{{env_id}}/addonsConfig'
+create_url: '{{env_id}}/addonsConfig:setAddonEnablement'
+update_url: '{{env_id}}/addonsConfig:setAddonEnablement'
+update_verb: 'POST'
+exclude_delete: true
```

What would you think about sending a POST request to unset AddonEnablement on delete? This would match [the behavior of AddonsConfig](https://github.com/GoogleCloudPlatform/magic-modules/blob/55ec2413bf9eee962665d452b46aa629e3252a49/mmv1/products/apigee/AddonsConfig.yaml#L28-L29).

```suggestion
delete_url: '{{env_id}}/addonsConfig:setAddonEnablement' 
delete_verb: 'POST'
```

---

It's not possible with this api because Delete operation test sends null object as input to api. AddonsConfig api allows null object as input but this api doesn't allow. 

```
sharmajai@jai-desktop:~/magic-modules$ curl -H "$AUTH" -H "Content-Type: application/json" -X POST "https://apigee.googleapis.com/v1/organizations/test-apigee-org:setAddons"
{
  "name": "organizations/test-apigee-org/operations/4296f43d-9fa1-4341-bd00-d8221852dafa",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.apigee.v1.OperationMetadata",
    "operationType": "UPDATE",
    "targetResourceName": "organizations/test-apigee-org",
    "state": "IN_PROGRESS"
  }
}
sharmajai@jai-desktop:~/magic-modules$ curl -H "$AUTH" "https://apigee.googleapis.com/v1/organizations/test-apigee-org/operations/4296f43d-9fa1-4341-bd00-d8221852dafa"
{
  "name": "organizations/test-apigee-org/operations/4296f43d-9fa1-4341-bd00-d8221852dafa",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.apigee.v1.OperationMetadata",
    "operationType": "UPDATE",
    "targetResourceName": "organizations/test-apigee-org",
    "state": "FINISHED"
  },
  "done": true
}
sharmajai@jai-desktop:~/magic-modules$ curl -H "$AUTH" -H "Content-Type: application/json" -X POST "https://apigee.googleapis.com/v1/organizations/test-apigee-org/environments/test/addonsConfig:setAddonEnablement"
{
  "error": {
    "code": 400,
    "message": "addon enablement request \u003cnil\u003e is not supported",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.RequestInfo",
        "requestId": "2385760879381416100"
      },
      {
        "@type": "type.googleapis.com/google.rpc.DebugInfo",
        "detail": "[ORIGINAL ERROR] generic::invalid_argument: addon enablement request \u003cnil\u003e is not supported [google.rpc.error_details_ext] { message: \"addon enablement request \u003cnil\u003e is not supported\" details { [type.googleapis.com/google.rpc.RequestInfo] { request_id: \"2385760879381416100\" } } }"
      }
    ]
  }
}
```

---

unfortunate! Thanks for verifying.